packages/artifact/src/internal/client.ts

```diff
@@ -39,6 +40,13 @@ export class Client implements ArtifactClient {
     rootDirectory: string,
     options?: UploadOptions | undefined
   ): Promise<UploadResponse> {
-    return uploadArtifact(name, files, rootDirectory, options)
+    try {
```

Comment:
SideNote, you could use the same `isGHES()` function early on here and immediately fail the upload We don't want anyone on GHES using this package yet and it will be a while until we add support for GHES.